test/build-test.js

```diff
@@ -67,11 +68,16 @@ describe('build', function() {
 			var s1 = delay(1, fromArray([1,2]));
 			var s2 = fromArray([3,4]);
 
-			return reduce(function(a, x) {
-				return a.concat(x);
-			}, [], build.concat(s1, s2)).then(function(a) {
-				expect(a).toEqual([1,2,3,4]);
-			});
+			var scheduler = new TestScheduler();
+			scheduler.tick(1);
```

Comment:
This is the gist of the TestScheduler.  Create one, and then call `tick` to arrange for it to run virtual timeline of a certain number of ticks.